Makefile

```diff
@@ -622,11 +622,15 @@ show:
 		echo "Error: show target needs REF= set" >&2; \
 	fi
 
+
+shellquote = '$(subst ','\'',$1)'#
+
 grep-tgt = $(GIT_REPOS:$(SRC_DIR)/%=%.grep)
 RE ?= $(filter-out grep $(grep-tgt), $(MAKECMDGOALS))
 .PHONY: grep $(grep-tgt)
 $(grep-tgt):
-	${Q}git -C $(@:%.grep=$(SRC_DIR)/%) grep "$(RE)" | sed "s#^#$(@:%.grep=$(SRC_DIR)\/%)/#"
+	@git -C $(call shellquote,$(@:%.grep=$(SRC_DIR)/%)) grep -nEe $(call shellquote,$(RE)) | sed 's#^#$(@:%.grep=$(SRC_DIR)/%)/#'
```

Comment:
1.  `-E` is change in behavior that is not mentioned in the commit message, and not really welcome one. _If_ you really need it, add `make egrep` for that. 

2. You don't need to call shellquote on the dir name.

Reply:
> 1. `-E` is change in behavior that is not mentioned in the commit message, and not really welcome one. _If_ you really need it, add `make egrep` for that.

Sorry about that.  Fixed by adding a GREPFLAGS parameter to allow passing arbitrary flags (not just `-E`).

> 2. You don't need to call shellquote on the dir name.

Done.